### PR TITLE
Adds ignores for IntelliJ IDEA generated files

### PR DESCRIPTION
**Pull request by dposada**
(and removes 3 unrelated duplicate entries)

### CODE REVIEW

**Comment by dposada** [comment]
@shamsimam can you please take a look?

File: .gitignore [hunk 1]
```diff
@@ -23,8 +22,8 @@ pom.xml.asc
 .lein-deps-sum
 .lein-failures
 .lein-plugins
-.lein-repl-history
-.nrepl-port
```
Comment (wyegelwel) [inline .gitignore]: These two lein ones seem as though they should stay in ignore. Why are you removing them?

**Comment by dposada** [comment]
Agree, they were duplicated elsewhere in the file

**Comment by wyegelwel** [comment]
It was really important to ignore them :)

**Comment by wyegelwel** [comment]
LGTM, will wait on @shamsimam to look as well

File: .gitignore (hunk 1 above)
Comment (shamsimam) [inline .gitignore]: I agree.

**Comment by shamsimam** [comment]
Can we sort the contents of the file?

**Comment by dposada** [comment]
@shamsimam .gitignore is now lexicographically sorted :)

**Pull request merged**